torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1158,13 +1159,16 @@ def _center_crop_compute_crop_anchor(
 
 def center_crop_image_tensor(image: torch.Tensor, output_size: List[int]) -> torch.Tensor:
     crop_height, crop_width = _center_crop_parse_output_size(output_size)
-    image_height, image_width = get_spatial_size_image_tensor(image)
+    shape = image.shape
+    if image.numel() == 0:
+        return image.reshape(shape[:-2] + (crop_height, crop_width))
+    image_height, image_width = shape[-2:]
 
     if crop_height > image_height or crop_width > image_width:
         padding_ltrb = _center_crop_compute_padding(crop_height, crop_width, image_height, image_width)
-        image = pad_image_tensor(image, padding_ltrb, fill=0)
+        image = _FT.torch_pad(image, _FT._parse_pad_padding(padding_ltrb), value=0.0)
```

Comment:
There is no real reason to hit `pad_image_tensor()` and go through the validation of input, parsing of parameters and multiple method calls to actually hit PyTorch's pad. We should try to be as explicit as possible on the internal implementations, as this pays out.

Here are V1 vs V2 benchmarks on latest main:
```
[-------------- CenterCrop cpu torch.float32 -------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   42 (+-  0) us  |   57 (+-  0) us
      (16, 3, 40, 50)  |  528 (+-  1) us  |  555 (+-  2) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   88 (+-  1) us  |  104 (+-  3) us
      (16, 3, 40, 50)  |  615 (+-  5) us  |  643 (+- 19) us

Times are in microseconds (us).

[------------- CenterCrop cuda torch.float32 -------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   48 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   48 (+-  0) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  1) us  |   48 (+-  1) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   48 (+-  1) us

Times are in microseconds (us).

[--------------- CenterCrop cpu torch.uint8 --------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   68 (+-  0) us  |   77 (+-  0) us
      (16, 3, 40, 50)  |  760 (+-  4) us  |  785 (+-  4) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |  106 (+-  1) us  |  116 (+-  2) us
      (16, 3, 40, 50)  |  837 (+-  6) us  |  861 (+- 18) us

Times are in microseconds (us).

[-------------- CenterCrop cuda torch.uint8 --------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   48 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   48 (+-  0) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  1) us  |   48 (+-  1) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   48 (+-  1) us

Times are in microseconds (us).
```

Here is after the PR:
```
[-------------- CenterCrop cpu torch.float32 -------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   45 (+-  0) us  |   42 (+-  0) us
      (16, 3, 40, 50)  |  525 (+-  2) us  |  535 (+-  1) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   88 (+-  3) us  |   88 (+-  1) us
      (16, 3, 40, 50)  |  615 (+- 23) us  |  626 (+- 18) us

Times are in microseconds (us).

[------------- CenterCrop cuda torch.float32 -------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   35 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   35 (+-  0) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   35 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   35 (+-  0) us

Times are in microseconds (us).

[--------------- CenterCrop cpu torch.uint8 --------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   68 (+-  0) us  |   65 (+-  0) us
      (16, 3, 40, 50)  |  763 (+-  4) us  |  773 (+-  6) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |  106 (+-  3) us  |  104 (+-  2) us
      (16, 3, 40, 50)  |  839 (+- 22) us  |  851 (+- 23) us

Times are in microseconds (us).

[-------------- CenterCrop cuda torch.uint8 --------------]
                       |        old       |        new     
1 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   34 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   35 (+-  0) us
6 threads: ------------------------------------------------
      (3, 40, 50)      |   37 (+-  0) us  |   35 (+-  0) us
      (16, 3, 40, 50)  |   31 (+-  0) us  |   35 (+-  1) us

Times are in microseconds (us).
```